test/e2e/event_transformation_test.go

```diff
@@ -0,0 +1,151 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/knative/eventing/test"
+	pkgTest "github.com/knative/pkg/test"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+/*
+TestEventTransformation tests the following scenario:
+
+EventSource ---> Channel ---> Subscriptions ---> Channel ---> Subscriptions ----> Service(Logger)
+                                   ^
+                                   |
+                                   |
+                                   |-----------> Service(Transformation)
+*/
+func TestEventTransformation(t *testing.T) {
+	clients, cleaner := Setup(t, t.Logf)
+
+	const (
+		senderName = "e2e-complexscen-sender"
+		msgPostfix = "######"
+	)
+
+	var channelNames = [2]string{"e2e-complexscen1", "e2e-complexscen2"}
```

Comment:
This setup part if pretty similar to the other test, why not encapsulating it in a helper shared function?

Reply:
Refactored some code into functions, like `CreatePodAndServiceReady` and `SendFakeEventToChannel`.